File: 36-valid-sudoku/valid-sudoku.java
class Solution 
{
    public boolean isValidSudoku(char[][] board) 
    {
        Set<String> set=new HashSet<>();

        for(int i=0;i<board.length;i++)
        {
            for(int j=0;j<board[i].length;j++)
            {
                char c=board[i][j];

                if(c!='.')
                {
                    if(!set.add("row"+c+i)||!set.add("col"+c+j)||!set.add("box"+c+i/3+j/3))
                    {
                        return false;
                    }
                }
            }
        }

        return true;   
    }
}

File: 36-valid-sudoku/valid-sudoku.kt
class Solution {
    fun isValidSudoku(board: Array<CharArray>): Boolean {
        val set=HashSet<String>()

        for(i in board.indices)
        {
            for(j in board[i].indices)
            {
                val c=board[i][j]

                if(c!='.')//number
                {
                    if(!set.add("row$i$c") || !set.add("col$j$c") || !set.add("box${i/3}${j/3}$c"))
                    {
                        return false
                    }
                }
            }
        }

        return true
    }
}